subprojects/core/src/main/java/org/gradle/api/internal/AbstractValidatingNamedDomainObjectContainer.java

```diff
@@ -38,7 +38,7 @@ protected AbstractValidatingNamedDomainObjectContainer(Class<T> type, Instantiat
     }
 
     public T create(String name, Action<? super T> configureAction) throws InvalidUserDataException {
-        NameValidator.validate(name);
+        NameValidator.validate(name, "name", "Please define a valid name in your build file");
```

Comment:
❌ The problem doesn't have to be in the build file, it could be anywhere. I don' think this additional sentence adds any value.